models/apple_search_ads__ad_group_report.sql

```diff
@@ -1,48 +1,57 @@
-with ad_group_report as (
+with report as (
 
     select *
     from {{ var('ad_group_report') }}
+), 
 
-), ad_group as (
+ad_group as (
 
     select * 
-    from {{ ref('int_apple_search_ads__most_recent_ad_group') }}
+    from {{ var('ad_group_history') }}
+    where is_most_recent_record = True
+), 
 
-), campaign as (
+campaign as (
 
     select *
-    from {{ ref('int_apple_search_ads__most_recent_campaign') }}
+    from {{ var('campaign_history') }}
+    where is_most_recent_record = True
+), 
 
-), organization as (
+organization as (
 
     select * 
     from {{ var('organization') }}
+), 
 
-), joined as (
+joined as (
 
     select 
-        ad_group_report.date_day,
+        report.date_day,
         organization.organization_id,
         organization.organization_name,
         campaign.campaign_id, 
         campaign.campaign_name, 
         ad_group.ad_group_id,
         ad_group.ad_group_name,
-        ad_group_report.currency,
-        ad_group_report.taps,
-        ad_group_report.new_downloads,
-        ad_group_report.redownloads,
-        (ad_group_report.new_downloads + ad_group_report.redownloads) as total_downloads,
-        ad_group_report.impressions,
-        ad_group_report.spend
-    from ad_group_report
+        report.currency,
```

Comment:
I notice there are a number of other fields in the `history` staging models (such as `status`) that we don't bring through to this final cte. Is there a reason we don't bring them through to this final model?

Reply:
Same question above for the other final models.

Reply:
Just added a few more of the history fields into each report! Some fields I didn't bring in such as `bid_amount` / `currency` just because it's an additional currency metric that I think is nice to have as a part of the history tables however, am unsure how folks would use it in reports so I opted to keep those out of the models. But have added all `*_status` fields & `end_at` & `start_at` fields that could be useful in the end reports!